Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp

```diff
@@ -3915,55 +3917,285 @@ JSGlobalContextRef webkit_web_view_get_javascript_global_context(WebKitWebView*
 }
 #endif
 
-static void webkitWebViewRunJavaScriptCallback(API::SerializedScriptValue* wkSerializedScriptValue, const ExceptionDetails& exceptionDetails, GTask* task)
+static void webkitWebViewRunJavaScriptWithParams(WebKitWebView* webView, RunJavaScriptParameters&& params, const char* worldName, GRefPtr<GTask>&& task)
 {
-    if (g_task_return_error_if_cancelled(task))
-        return;
+    auto world = worldName ? API::ContentWorld::sharedWorldWithName(String::fromUTF8(worldName)) : Ref<API::ContentWorld> { API::ContentWorld::pageContentWorld() };
+    getPage(webView).runJavaScriptInFrameInScriptWorld(WTFMove(params), std::nullopt, world.get(), [task = WTFMove(task)] (auto&& result) {
+        if (g_task_return_error_if_cancelled(task.get()))
+            return;
 
-    if (!wkSerializedScriptValue) {
-        StringBuilder builder;
-        if (!exceptionDetails.sourceURL.isEmpty()) {
-            builder.append(exceptionDetails.sourceURL);
-            if (exceptionDetails.lineNumber > 0)
-                builder.append(':', exceptionDetails.lineNumber);
-            if (exceptionDetails.columnNumber > 0)
-                builder.append(':', exceptionDetails.columnNumber);
-            builder.append(": ");
+        if (result.has_value()) {
+            if (!result.value())
+                g_task_return_new_error(task.get(), WEBKIT_JAVASCRIPT_ERROR, WEBKIT_JAVASCRIPT_ERROR_INVALID_RESULT, "Unsupported result type");
+            else {
+                g_task_return_pointer(task.get(), webkitJavascriptResultCreate(result.value()->internalRepresentation()),
+                    reinterpret_cast<GDestroyNotify>(webkit_javascript_result_unref));
+            }
+        } else {
+            ExceptionDetails exceptionDetails = WTFMove(result.error());
+            StringBuilder builder;
+            if (!exceptionDetails.sourceURL.isEmpty()) {
+                builder.append(exceptionDetails.sourceURL);
+                if (exceptionDetails.lineNumber > 0)
+                    builder.append(':', exceptionDetails.lineNumber);
+                if (exceptionDetails.columnNumber > 0)
+                    builder.append(':', exceptionDetails.columnNumber);
```

Comment:
I know this should surely never happen, but if somehow we wind up with a column number but no line number, then this will print something very confusing, so better:

```suggestion
                if (exceptionDetails.lineNumber > 0) {
                    builder.append(':', exceptionDetails.lineNumber);
                    if (exceptionDetails.columnNumber > 0)
                        builder.append(':', exceptionDetails.columnNumber);
                }
```

Reply:
Good point